File: java/1-sequence-struct/basic-input-output/src/Main.java
// Write a program that asks the user to enter first name, age, gender, height. Print the information.

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        // Instantiating object in java for data entry
        Scanner sc = new Scanner(System.in);

        int age;
        char gender;
        double height;
        String name;

        System.out.print("Enter the first name: ");
        name = sc.nextLine();
        System.out.print("Enter age: ");
        age = sc.nextInt();
        System.out.print("Enter the gender(M/F): ");
        gender = sc.next().charAt(0);
        System.out.print("Enter the height: ");
        height = sc.nextDouble();

        System.out.printf("\nName: %s\nAge: %d\nGender: %c\nHeight: %.2f\n", name, age, gender, height);

    }
}

File: c/1-sequence-struct/basic-input-output/main.c
// Write a program that asks the user to enter first name, age, gender, height. Print the information.

#include <stdio.h>
#include <stdlib.h>

/* Function to clear the input buffer, due to possible problems reading entire data after text input,
where the line break is left in the buffer, affecting the next read.*/ 
void clearBuffer() {
    int c;
    while ((c = getchar()) != '\n' && c != EOF);
}

int main()
{
    int age;
    char name[20], gender;
    double height;

    printf("Enter first name: ");
    scanf("%s", name);
    printf("Enter age: ");
    scanf("%d", &age);
    printf("Enter the gender(M/F): ");
    clearBuffer();
    scanf("%c", &gender);
    printf("Enter the height: ");
    scanf("%lf", &height);

    printf("\nName: %s\nAge: %d\nGender: %c\nHeight: %.2lf\n", name, age, gender, height);
    return 0;
}
